tools/mkfs.c

```diff
@@ -419,6 +626,12 @@ int main(int argc, char **argv)
         if (nr < 0) {
             halt("couldn't read from boot image file %s: %s\n", bootimg_path, strerror(errno));
         }
+
+        /* The on-disk kernel log dump section is immediately before the first partition. */
+        offset += KLOG_DUMP_SIZE;
+
+        if (uefi_loader)
+            offset = write_uefi_part(out, offset, uefi_loader);
```

Comment:
If the hypervisor does not support UEFI and image was built with UEFI, would it still work? i.e. the legacy-BIOS bits are still written to the image.

* if so, would it make sense to always include UEFI support (and add option to skip UEFI if it makes a a significant difference in size)?
* if not, would it make sense to make it work with both, legacy-BIOS and UEFI, modes (like modern OS support)?

Reply:
Yes, an image built with UEFI will still work on hypervisors that don't support UEFI.
I'm still working on adding the required functionality in Ops, but the idea is to include UEFI support only when needed, since it adds 33 MB to the image size.

Reply:
Great! and wow, 33MB is significant enough. 😁
(must be the bare minimum size to support UEFI)

I think in ops, `hyper-v2` and `azure2` providers could be (additionally) added for users to explicitly choose gen1 or gen2 of Hyper-V VMs.

Reply:
In https://github.com/nanovms/ops/pull/981, I opted to not define additional cloud providers, but add handling of the `ImageType` CloudConfig attribute (and for Hyper-V the `Flavor` CloudConfig attribute as well).